python/ray/autoscaler/_private/resource_demand_scheduler.py

```diff
@@ -592,8 +590,12 @@ def reserve_and_allocate_spread(
             unfulfilled, including_reserved = get_bin_pack_residual(
                 new_node_resources, unfulfilled, strict_spread=True
             )
-            assert not unfulfilled
-            node_resources += including_reserved
+            if unfulfilled:
+                logger.debug("Unfulfilled strict placement group: {}".format(bundles))
+                continue
+            _inplace_add(node_type_counts, to_launch)
+            _inplace_add(to_add, to_launch)
+            node_resources = updated_node_resources + including_reserved
```

Comment:
After we are sure that the strict spread bundle can be scheduled with new nodes, we update the `node_resources`, `node_type_counts`, and `to_add`.